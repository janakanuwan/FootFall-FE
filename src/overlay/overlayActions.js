import { createAction } from 'redux-actions';

import { HIDE_OVERLAY, SHOW_OVERLAY } from "../const/action-types";

const showOverlay = createAction(
  SHOW_OVERLAY,
  overlayData => overlayData,
);

const hideOverlay = createAction(
  HIDE_OVERLAY,
  overlayData => overlayData,
);

export { showOverlay, hideOverlay };
