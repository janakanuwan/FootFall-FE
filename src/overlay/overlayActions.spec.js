import { hideOverlay, showOverlay } from './overlayActions';

describe('overlayActions', () => {

  it('should create an action to show an overlay', () => {
    expect(showOverlay({ id: 'user_login', title: 'Loading', message: 'Validating' })).toMatchSnapshot();
  });

  it('should create an action to hide an overlay', () => {
    expect(hideOverlay({ id: 'user_login', title: 'Loading', message: 'Validating' })).toMatchSnapshot();
  });
});
