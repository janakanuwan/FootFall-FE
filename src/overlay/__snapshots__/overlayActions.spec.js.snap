// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlayActions should create an action to hide an overlay 1`] = `
Object {
  "payload": Object {
    "id": "user_login",
    "message": "Validating",
    "title": "Loading",
  },
  "type": "HIDE_OVERLAY",
}
`;

exports[`overlayActions should create an action to show an overlay 1`] = `
Object {
  "payload": Object {
    "id": "user_login",
    "message": "Validating",
    "title": "Loading",
  },
  "type": "SHOW_OVERLAY",
}
`;
